react/components/molecules/keyboard/keyboard.js

```diff
@@ -39,13 +41,52 @@ export class Keyboard extends Component {
         super(props);
 
         this.state = {
-            symbolicKeyboard: this.props.symbolicKeyboard
+            symbolicKeyboard: this.props.symbolicKeyboard,
+            animationPositionY: new Animated.Value(212),
+            animationOpacity: new Animated.Value(0)
         };
 
         this.keys = "1234567890qwertyuiopasdfghjklzxcvbnm";
         this.keyColumns = [10, 10, 9, 9];
     }
 
+    componentDidMount() {
+        this.showKeyboardAnimated();
+    }
+
+    showKeyboardAnimated = async () => {
+        if (!this.props.animations) return;
+
+        await this._animateKeyboard(0, 1);
+    };
+
+    hideKeyboardAnimated = async () => {
+        if (!this.props.animations) return;
+
+        await this._animateKeyboard(212, 0, 200);
+    };
+
+    _animateKeyboard = async (positionY, opacity, duration = 250) => {
+        return new Promise(resolve => {

```

Comment:
why the promise? I think you can just to the start of the animation and all the functions that call this don't need the await, since there is no action that is dependent on the end of the animation

Reply:
because I want to know when the hideKeyboardAnimated animation stops and being able to await for the callback to move to the next tick

Reply:
Here's how the implementation looks on personalization page
```javascript


    onCloseKeyboardPress = async () => {
        await this.keyboardRef.hideKeyboardAnimated();
        this.setState({
            keyboardVisible: false
        });
    };
```

Reply:
Got it, nice :+1: 
The only thing is to await `this.showKeyboardAnimated();` in componentDidMount